src/world/area_mgm/mgm_00/E0F980.c

```diff
@@ -1,9 +1,139 @@
+        } else {
+            draw_msg(MESSAGE_ID(0x8,0x1D), 64, 57, alpha, MSG_PAL_TEAL, 0);
+            draw_number(gPlayerData.smashGamePlays,  219, 78,  TRUE, MSG_PAL_WHITE, alpha, 3);
+            draw_number(gPlayerData.smashGameTotal,  219, 93,  TRUE, MSG_PAL_WHITE, alpha, 3);
+            draw_number(gPlayerData.smashGameRecord, 219, 108, TRUE, MSG_PAL_WHITE, alpha, 3);
+        }
+
+        draw_msg(MESSAGE_ID(0x8,0x1E), 58,  78,  alpha, MSG_PAL_WHITE, DRAW_MSG_STYLE_MENU);
+        draw_msg(MESSAGE_ID(0x8,0x23), 223, 78,  alpha, MSG_PAL_WHITE, DRAW_MSG_STYLE_MENU);
+        draw_msg(MESSAGE_ID(0x8,0x1F), 58,  93,  alpha, MSG_PAL_WHITE, DRAW_MSG_STYLE_MENU);
+        draw_msg(MESSAGE_ID(0x8,0x21), 223, 93,  alpha, MSG_PAL_WHITE, DRAW_MSG_STYLE_MENU);
+        draw_msg(MESSAGE_ID(0x8,0x20), 58,  108, alpha, MSG_PAL_WHITE, DRAW_MSG_STYLE_MENU);
+        draw_msg(MESSAGE_ID(0x8,0x21), 223, 108, alpha, MSG_PAL_WHITE, DRAW_MSG_STYLE_MENU);
+    }
+}
+
+void N(animate_and_draw_record)(void* renderData) {
+    RecordDisplayData* data = (RecordDisplayData*)evt_get_variable(NULL, GW(RECORD_DISPLAY_MAP_VAR));
+
+    switch (data->state) {
+    case RECORD_START_SHOW:
+        data->stateTimer = 0;
+        data->state = RECORD_STATE_FADE_IN;
+        snd_start_sound(SOUND_21C, 0, 0);
+        // fallthrough
+    case RECORD_STATE_FADE_IN:
+        data->stateTimer++;
+        data->alpha = update_lerp(EASING_LINEAR, 32.0f, 255.0f, data->stateTimer, FADE_IN_TIME);
+        data->curAlpha = data->alpha;
+        if (data->stateTimer >= FADE_IN_TIME) {
+            data->state = RECORD_STATE_VISIBLE;
+            data->alpha = 255;
+            data->curAlpha = 255;
+        }
+        break;
+    case RECORD_STATE_VISIBLE:
+        data->alpha = 255;
+        data->curAlpha = 255;
+        if (gGameStatusPtr->currentButtons & (BUTTON_A | BUTTON_B)) {
+            data->state = RECORD_START_HIDE;
+        }
+        if (data->state != RECORD_START_HIDE) {
+            break;
+        }
+        // fallthrough
+    case RECORD_START_HIDE:
+        data->stateTimer = 0;
+        data->state = RECORD_STATE_FADE_OUT;
+        snd_start_sound(SOUND_MENU_BACK, 0, 0);
+        // fallthrough
+    case RECORD_STATE_FADE_OUT:
+        data->stateTimer++;
+        data->alpha = update_lerp(EASING_LINEAR, data->curAlpha, 0.0f, data->stateTimer, FADE_OUT_TIME);
+        if (data->stateTimer >= FADE_OUT_TIME) {
+            data->state = RECORD_STATE_DONE;
+        }
+        break;
+    }
+    N(draw_record_display)(data, data->alpha);
+}
+
+void N(work_draw_record)(void) {
+    RenderTask task;
+
+    task.renderMode = RENDER_MODE_2D;
+    task.appendGfxArg = 0;
+    task.appendGfx = &N(animate_and_draw_record);
+    task.distance = 0;
+
+    queue_render_task(&task);
+}
+
+ApiStatus N(UpdateRecordDisplay)(Evt* script, s32 isInitialCall) {
+    RecordDisplayData* data;
```

Comment:
technically, despite it not being strictly necessary here, I think we want to start declaring Bytecode* args = script->ptrReadPos in any script func that uses it for consistency + so we could add more var reads later potentially